packages/analytics-core/src/diagnostic.ts

```diff
@@ -0,0 +1,82 @@
+import { Diagnostic as IDiagnostic, Event, Result } from '@amplitude/analytics-types';
+import { DIAGNOSTIC_ENDPOINT } from './constants';
+
+interface DiagnosticEvent {
+  time: number;
+  event_properties: {
+    response_error_code: number;
+    trigger: string;
+    action: string;
+    event_count: number;
+  };
+  library: string;
+}
+
+export class Diagnostic implements IDiagnostic {
+  serverUrl: string;
+  queue: DiagnosticEvent[] = [];
+
+  private scheduled: ReturnType<typeof setTimeout> | null = null;
+  // deault delay is 1 minute
+  // make it private to prevent users from changing it to smaller value
+  private delay = 60000;
+
+  constructor(serverUrl?: string) {
```

Comment:
To make the config easier to maintain (expand) can you please create an DiagnosticsOptions object for config vs constructor args.